### PR TITLE
CI: enable Prettier checking

### LINKED ISSUE #1063: CI: check markdown formatting and common gotchas
Now that we starting to have more and more contributors, the project would benefit from standardizing Markdown formatting, and a prettifier and linter could help in this respect.

- I've (@chalin) started experimenting with [Prettier](https://prettier.io), e.g., see:
  - #1290
  - https://github.com/open-telemetry/opentelemetry.io/pulls?q=is%3Apr+prettier+is%3Aclosed
- Note that there's module for formatting Go templates. [Read more](https://discourse.gohugo.io/t/formatter-for-go-templates/38403)

---

- [x] #2286
- [x] #2292
- [x] #2302
- [x] #2309
- [x] #2311
- [x] #2315
- [x] #2316
- [x] #2338
- [x] #2363
- [x] #2368
- [x] #2377
- [x] #2372
- [x] Go docs
  - [x] #2394
  - [x] https://github.com/open-telemetry/opentelemetry-go/pull/3762
  - [x] #2380
- [x] #2389
- [x] #2390
- [x] #2392
- [x] #2393
- [x] #2398
- [x] #2399
- [x] #2408
- [x] #2412
- [x] Community pages: #2414
- [x] Data files and more: #2414

### PR DESCRIPTION
**Pull request by chalin**
- Enables Prettier checking over all non-ignored resources to CI by creating NPM scripts that contributors can run too
- Closes #2179 by superseding it
- Closes #1063, though we'll have to progressively relax the `.prettierignore` rules

✅ **Note that this PR was seeded with a formatting issue** in the `README.md` file. The build did fail (for details, see the comment below). ~If the build fails, then that will confirm that the changes in this PR are working. I'll then remove the `README.md` file change.~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,8 +6,7 @@ and blog. The site is [built][contributing.md] using [Hugo][] and hosted on
 
 ## Get involved
 
-If you are new to OpenTelemetry and just get started with it, you are in a
-perfect position to help us get better: the website and documentation is the
+If you are new to OpenTelemetry and just get started with it, you are in a perfect position to help us get better: the website and documentation is the
```
Comment (chalin) [inline README.md:9]: ⚠️ This is seeding a formatting issue to see if Prettier will cause the build to fail.

File: package.json [hunk 2]
```diff
@@ -53,7 +55,8 @@
   },
   "private": true,
   "prettier": {
-    "proseWrap": "always"
+    "proseWrap": "always",
+    "singleQuote": true
```
Comment (chalin) [inline package.json:59]: I prefer the use of single quotes when possible, hence this new option (FYI).

Comment (svrnm) [reply package.json:59]: +1

File: README.md (hunk 1 above)
Comment (chalin) [reply README.md]: This got detected as expected.

**Review by cartermp — APPROVED** [review]

**Pull request merged**